nixos/modules/services/misc/sr-ht/git.nix

```diff
@@ -0,0 +1,115 @@
+        default = "redis://localhost:6379/1";
+        description = ''
+          The redis connection used for the Celery worker.
+        '';
+      };
+
+      postUpdateScript = mkOption {
+        type = types.path;
+        default = "${git-sr-ht}/bin/gitsrht-update-hook";
+        description = ''
+          A post-update script which is installed in every git repo.
+        '';
+      };
+
+      # TODO: Assertion
+      oauth-client.id = mkOption {
+        type = types.either types.int types.str;
+        default = "CHANGEME";
+        description = ''
+          OAuth client ID for meta.sr.ht from git.sr.ht, register your client at
+          meta.example.org/oauth.
+        '';
+      };
+
+      # TODO: Assertion
+      oauth-client.secret = mkOption {
+        type = types.str;
+        default = "CHANGEME";
+        description = ''
+          OAuth client secret for meta.sr.ht from git.sr.ht, register your client at
+          meta.example.org/oauth.
+        '';
+      };
+
+      # TODO: Better default?
+      repos = mkOption {
+        type = types.path;
+        default = "/var/lib/git";
+        description = ''
+          Path to git repositories on disk.
+        '';
+      };
+
+      dispatch = mkOption {
+        type = types.listOf types.str;
+        default = [ "${cfg.packages.git}/bin/gitsrht-keys=git:git" ];
+        example = [ "${cfg.packages.git}/bin/gitsrht-keys=git:git" "${cfg.packages.git}/bin/man-srht-keys=man:man" ];
+        description = ''
+          The authorized keys hook uses this to dispatch to various handlers
+          The format is a program to exec into as the key, and the user to match as the
+          value. When someone tries to log in as this user, this program is executed
+          and is expected to omit an AuthorizedKeys file.
+        '';
+      };
+    };
+  };
+
+  config = mkIf (cfg.enable && (elem "git" cfg.services)) {
+    # post_upgrade() {
+    #   gitsrht-migrate -a upgrade head
+    # }
+
+    # Type, Name, ID, GECOS, Home Directory, Shell
+    # u gitsrht - "git.sr.ht user"
+    # Type u: create user, group if doesn't exist, group name = user name
+    # ID -: allocate automatically
+
+    users = {
+      users = [
+        { name = gcfg.user;
+          group = gcfg.user;
+          description = "git.sr.ht user"; } ];
+      groups = [
+        { name = gcfg.user; } ];
+    };
+
+    services = {
+      nginx.virtualHosts."git.sr.ht".locations = {
+        "/".proxyPass = "http://${cfg.address}:${toString gcfg.port}";
+        "/static".root = "${git-sr-ht}/${git-sr-ht.sitePackages}/gitsrht";
+      };
+    };
+
+    systemd.services."git.sr.ht" = {
+      after = [ "redis.service" "postgresql.service" "network.target" ];
+      requires = [ "redis.service" "postgresql.service" ];
+      wantedBy = [ "multi-user.target" ];
+
+      description = "git.sr.ht website service";
+      path = with pkgs; [ (python36Packages.gunicorn.overridePythonAttrs (oldAttrs: {
```

Comment:
It should be sufficient to use `python36.withPackages`.

Reply:
`python36.withPackages` doesn't seem to work with the `*-sr-ht` packages.

Reply:
Please be more specific.

Reply:
Makes sense since meta-sr-ht is using `buildPythonApplication`.